### PR TITLE
SignerListSet and SetRegularKey

### PR DESCRIPTION
**Pull request by florent-uzio**
# Title

This PR aims to add the `Validate` and `Flatten` function to `SetRegularKey` and `SignerListSet`.
The tests have 100% coverage.
The PR also adds comments for the SignerList entry and the relevant `Flatten` functions.

### CODE REVIEW

**Comment by florent-uzio** [comment]
@GuillemGarciaDev , gentle reminder about this PR as well

File: xrpl/ledger-entry-types/signer_list.go [hunk 1]
```diff
@@ -8,27 +8,108 @@ const (
...
-	OwnerNode         string
-	SignerEntries     []SignerEntryWrapper
-	SignerListID      uint64
-	SignerQuorum      uint64
+	// A hint indicating which page of the owner directory links to this object, in case the directory consists of multiple pages.
+	OwnerNode string
+	// An array of Signer Entry objects representing the parties who are part of this signer list.
+	SignerEntries []SignerEntryWrapper
+	// An ID for this signer list. Currently always set to 0. If a future amendment allows multiple signer lists for an account, this may change.
+	SignerListID uint64
```
Comment (GuillemGarciaDev) [inline xrpl/ledger-entry-types/signer_list.go:62]: It is an `uint32`

File: xrpl/ledger-entry-types/signer_list.go [hunk 2]
```diff
@@ -8,27 +8,108 @@ const (
...
+	// A target number for signer weights. To produce a valid signature for the owner of this SignerList, the signers must provide valid signatures whose weights sum to this value or more.
+	SignerQuorum uint64
```
Comment (GuillemGarciaDev) [inline xrpl/ledger-entry-types/signer_list.go:64]: `uint32`

File: xrpl/ledger-entry-types/signer_list.go [hunk 3]
```diff
@@ -8,27 +8,108 @@ const (
...
 type SignerList struct {
-	LedgerEntryType   LedgerEntryType
-	Flags             SignerListFlags
-	PreviousTxnID     string
+	// The value 0x0053, mapped to the string SignerList, indicates that this is a SignerList ledger entry.
+	LedgerEntryType LedgerEntryType
+	// The identifying hash of the transaction that most recently modified this object.
+	PreviousTxnID string
+	// The index of the ledger that contains the transaction that most recently modified this object.
 	PreviousTxnLgrSeq uint64
```
Comment (GuillemGarciaDev) [inline xrpl/ledger-entry-types/signer_list.go:56]: `uint32`

File: xrpl/ledger-entry-types/signer_list.go
Comment (GuillemGarciaDev) [inline xrpl/ledger-entry-types/signer_list.go:1]: we should add the `Flatten` method for `SignerList`

File: xrpl/ledger-entry-types/signer_list.go [hunk 4]
```diff
@@ -8,27 +8,108 @@ const (
...
 }
 
+// Wrapper for SignerEntry
 type SignerEntryWrapper struct {
 	SignerEntry SignerEntry
 }
 
+// Flatten returns a map of the SignerEntryWrapper object
+func (s *SignerEntryWrapper) Flatten() FlatLedgerObject {
+	if s.SignerEntry != (SignerEntry{}) {
```
Comment (GuillemGarciaDev) [inline xrpl/ledger-entry-types/signer_list.go:74]: This condition is no neeeded

File: xrpl/transaction/signer_list_set.go [hunk 5]
```diff
@@ -1,20 +1,128 @@
...
+		return false, err
+	}
+
+	// All other checks are for if SignerQuorum is greater than 0
+	if s.SignerQuorum == 0 {
+		return true, nil
+	}
+
+	// Check if SignerEntries has at least 1 entry and no more than 32 entries
+	if len(s.SignerEntries) < MIN_SIGNERS || len(s.SignerEntries) > MAX_SIGNERS {
```
Comment (GuillemGarciaDev) [inline xrpl/transaction/signer_list_set.go:102]: rename all uppercase constants to camelcase

**Review by GuillemGarciaDev — CHANGES_REQUESTED** [review]

**Review by GuillemGarciaDev — APPROVED** [review]

**Pull request merged**